modules/tinymce/src/core/main/ts/keyboard/Nbsps.ts

```diff
@@ -149,22 +162,35 @@ const normalizeNbspInMiddleOfTextNode = (node: Text): boolean => {
   }
 };
 
-const normalizeNbspAtEnd = (root: SugarElement<Node>, node: Text): boolean => {
+const normalizeNbspAtEnd = (root: SugarElement<Node>, node: Text, makeNbsp: boolean): boolean => {
   const text = node.data;
   const lastPos = CaretPosition(node, text.length - 1);
-  if (isNbspAt(text, text.length - 1) && !needsToBeNbsp(root, lastPos)) {
+  if (!makeNbsp && isNbspAt(text, text.length - 1) && !needsToBeNbsp(root, lastPos)) {
     node.data = text.slice(0, -1) + ' ';
     return true;
+  } else if (makeNbsp && isWhiteSpaceAt(text, text.length - 1) && needsToBeNbspRight(root, lastPos)) {
+    node.data = text.slice(0, -1) + Unicode.nbsp;
+    return true;
   } else {
     return false;
   }
 };
 
-const normalizeNbsps = (root: SugarElement<Node>, pos: CaretPosition): Optional<CaretPosition> => Optional.some(pos).filter(hasNbsp).bind((pos) => {
-  const container = pos.container() as Text;
-  const normalized = normalizeNbspAtStart(root, container) || normalizeNbspInMiddleOfTextNode(container) || normalizeNbspAtEnd(root, container);
-  return normalized ? Optional.some(pos) : Optional.none();
-});
+const normalizeNbsps = (root: SugarElement<Node>, pos: CaretPosition): Optional<CaretPosition> => {
+  if (hasNbsp(pos)) {
```

Comment:
Should this also have the `NodeType.isText(pos.container())` check? It seems quite odd we'd have it in one place but not both... especially given we cast it as a text node below.

Reply:
Maybe this needs an early exit function before all this? For example:
```
const container = pos.container();
if (!NodeType.isText(container)) {
  return Optional.none();
}
```
That would also mean you can remove the casts if we do something like that.

Reply:
The `hasNbsp`-function already checks for if it is a text or not, that's why it wasn't included. But yeah, I can add that check as you suggest.